### DISCUSSION TITLE
tron sign TriggerSmartContract

### DISCUSSION THREAD

**Discussion by Binggggggggg** [WalletConnect/walletconnect-monorepo#2378] [Developer Support]
I use WalletConnectV2 to connect to Tron with my wallet
I want to connect with SunSwap to sign, but there should be a problem with the format,
sunswap rejects my signature, but i don't know what's the problem
My raw file uploaded to the chain is successful

```
{"result":{"raw_data":{"contract":[{"parameter":{"type_url":"type.googleapis.com/protocol.TriggerSmartContract","value":{"call_value":1000000,"contract_address":"416e0617948fe030a7e4970f8389d4ad295f249b7e","data":"7ff36000000000000000000000000000000000000000000000000000000080000000000000000000000000b94d22b4dd96571c566de4332df44062b93314f4000000000000000000000000000000000000000000000000000000006461d65c0000000009c7f77e6ded13c","owner_address":"41b94d22b4dd96571c566de4332df44062b93314f4"}},"type":"TriggerSmartContract"}],"expiration":1684169473000,"fee_limit":250000000,"ref_block_bytes":"d640","ref_block_hash":"1be86088f1cde905","timestamp":1684133473000},"signature":["1197bdd1caca805352c622c1d20be25c885a2419504aec40801"],"txID":"06cdc537b55e7405020ad8c02c869c94"}}
```

**Comment by finessevanes** [comment]
Hey - could you share your implementation and what the error is?

**Comment by Binggggggggg** [comment]
This is the data structure returned after I signed with sunswap, please
check if there are any missing parameters

vanes ***@***.***> 於 2023年5月25日 週四 下午12:45寫道：

**Comment by qixiaobro** [comment]
How did you use WalletConnectV2 to connect to the Tron network? I tried using the official demo but it doesn't support the Tron network. Could you please tell me how you achieved this? Thank you.

**Comment by Binggggggggg** [comment]
I am a developer of the mobile terminal. For me, the connection process is
the same as that of the Ethereum chain. Listen to
"Sign.instance.sessionProposalPublisher", and return the address and data
back through "Sign.instance.approve" to complete the connection

stephzhu ***@***.***> 於 2023年6月20日 週二 下午4:39寫道：
